samples/gibbs_ensemble/run_sim.py

```diff
@@ -0,0 +1,524 @@
+# (at your option) any later version.
+#
+# ESPResSo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Simulate a Gibbs-ensemble of a Lennard-Jones fluid at a fixed temperature.
+This script does the Monte-Carlo part of the Gibbs-ensemble, however for the
+energy calculation of the systems, two instances of the Client class in
+:file:`gibbs_ensemble.py` initialized (and possibly updated) in
+:file:`gibbs_client_system.py` are started which each run a different instance
+of ESPResSo.
+
+The Gibbs-ensemble implemented in these scripts closely refers to chapter 8 of
+Daan Fenkel and Berend Smit 'Understanding Molecular Simulation, Second edition'.
+All equation references noted in this script can be found there.
+
+Due to the cutoff and shifting of the LJ-potential the simulated points in the
+phase diagram differ from the long range uncut LJ-potential. The phase diagram
+of the used potential can be found in 'B. Smit, J. Chem. Phys. 96 (11), 1992,
+Phase diagrams of Lennard-Jones fluids'.
+"""
+
+from multiprocessing import Pipe
+import numpy as np
+from tqdm import tqdm
+import gzip
+import pickle
+import argparse
+import logging
+from enum import Enum, unique
+
+from gibbs_ensemble import MessageID
+from client_system import Gibbs_Client
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('T', type=float, help="Temperature of the system")
+parser.add_argument(
+    '-N',
+    type=int,
+    default=256,
+    help="Number of particles, default 256")
+parser.add_argument(
+    '--density',
+    type=float,
+    default=0.3,
+    help="Init density, default 0.3")
+parser.add_argument(
+    '--steps',
+    '-s',
+    type=int,
+    default=100000,
+    help="Number of steps, default is 1e5")
+parser.add_argument(
+    '--warmup',
+    '-w',
+    type=int,
+    default=5000,
+    help="Warmup steps, default 5000")
+parser.add_argument(
+    '--seed',
+    type=int,
+    default=17,
+    help="Random seed")
+parser.add_argument(
+    '--log',
+    action='store_true',
+    help="Pipe information to log file instead of stdout")
+parser.add_argument(
+    '--debug',
+    action='store_true',
+    help="Set for debugging info (use with small step number)")
+args = parser.parse_args()
+
+
+@unique
+class Moves(Enum):
+    # Particle moves
+    MOVE_PARTICLE = 1
+    EXCHANGE_VOLUME = 2
+    EXCHANGE_PARTICLE = 3
+
+
+# Number of client processes
+NUM_OF_CLIENTS = 2
+
+# Global number of particles
+GLOBAL_NUM_PART = args.N
+try:
+    assert((GLOBAL_NUM_PART % 2) == 0)
+except AssertionError:
+    raise TypeError("Wrong number of particles, must be even")
```

Comment:
The error message can be written in the assertion itself. The syntax is `assert EXPRESSION, MESSAGE`. Note there are no round braces, because `assert` is not a function. Adding round braces can lead to nasty bugs, here is an illustration:
```python
def foo(run_dangerous_function):
   assert run_dangerous_function, "You shouldn't run a dangerous function"
   drop_table()

def bar(run_dangerous_function):
   assert(run_dangerous_function, "You shouldn't run a dangerous function")
   drop_table()

foo(False) # safe: the assertion caught the False
bar(False) # oops!
```